Sources/Amplitude/DefaultTrackingOptions.swift

```diff
@@ -10,54 +10,57 @@ public class DefaultTrackingOptions {
     }
 
     public var sessions: Bool {
-        get {
-            autocaptureOptions.contains(.sessions)
-        }
-        set {
-            if newValue {
-                autocaptureOptions.insert(.sessions)
-            } else {
-                autocaptureOptions.remove(.sessions)
-            }
+        didSet {
+            delegate?.didChangeOptions(options: self)
         }
     }
 
     public var appLifecycles: Bool {
-        get {
-            autocaptureOptions.contains(.appLifecycles)
-        }
-        set {
-            if newValue {
-                autocaptureOptions.insert(.appLifecycles)
-            } else {
-                autocaptureOptions.remove(.appLifecycles)
-            }
+        didSet {
+            delegate?.didChangeOptions(options: self)
         }
     }
 
     public var screenViews: Bool {
-        get {
-            autocaptureOptions.contains(.screenViews)
-        }
-        set {
-            if newValue {
-                autocaptureOptions.insert(.screenViews)
-            } else {
-                autocaptureOptions.remove(.screenViews)
-            }
+        didSet {
+            delegate?.didChangeOptions(options: self)
         }
     }
 
-    var autocaptureOptions: AutocaptureOptions
+    weak var delegate: DefaultTrackingOptionsDelegate?
+
+    var toAutocaptureOptions: AutocaptureOptions {
+        return [
+            sessions ? .sessions : [],
+            appLifecycles ? .appLifecycles : [],
+            screenViews ? .screenViews : []
+        ].reduce(into: []) { $0.formUnion($1) }
+    }
 
     public init(
         sessions: Bool = true,
         appLifecycles: Bool = false,
         screenViews: Bool = false
     ) {
-        self.autocaptureOptions = []
         self.sessions = sessions
         self.appLifecycles = appLifecycles
         self.screenViews = screenViews
     }
+
+    convenience init(with delegate: DefaultTrackingOptionsDelegate) {
+        self.init()
+        self.delegate = delegate
+    }
+}
+
+protocol DefaultTrackingOptionsDelegate: AnyObject {
+    @available(*, deprecated)
+    func didChangeOptions(options: DefaultTrackingOptions)
+}
+
+extension Configuration: DefaultTrackingOptionsDelegate {
+    @available(*, deprecated)
+    func didChangeOptions(options: DefaultTrackingOptions) {
+        autocapture = options.toAutocaptureOptions
+    }
 }
```

Comment:
The beauty of a delegate is that the delegating class has no knowledge of its actual delegate. Why does this reference configuration here?

Reply:
I see, I will move it to the `Configuration` file to make it more aligned with the pattern.